### ISSUE TITLE
Requesting logs for Add-ons fails

### ISSUE THREAD

**Issue by alsFC** [home-assistant/supervisor#5382]
### Describe the issue you are experiencing

The log-view of any add-on fails after updating Supervisor to 2024.10.4 on the add-on panel.
Under System -> Logs I can see the logs of each add-on when selecting one.

### What type of installation are you running?

Home Assistant OS 13.2
Core 2024.10.4
Supervisor 2024.10.4
Frontend 20241002.4

### Which operating system are you running on?

Home Assistant Operating System

### Steps to reproduce the issue

1. Go to any Add-on
2. Select the Log panel
![Bildschirmfoto 2024-10-31 um 08 10 51](https://github.com/user-attachments/assets/ba8536d5-f567-4476-bebd-6beea86f7edd)

### Anything in the Supervisor logs that might be useful for us?

```txt
nothing regarding this in core- or supervisor-logs.
```

### System Health information

no repairs

**Comment by bcutter** [comment]
Updated from Supervisor 2024.10.3 to 2024.11.2.

Since then, errors when trying to read supervisor or addon logs, e. g.:

```
Failed to get a0d7b954_ssh logs, e.callApiRaw is not a function. (In 'e.callApiRaw("GET",`hassio/${t.includes("_")?`addons/${t}`:t}/logs${0!==r?`/boots/${r}`:""}`,void 0,i?{Range:i}:void 0)', 'e.callApiRaw' is undefined)
```

![image](https://github.com/user-attachments/assets/adea1407-3488-474e-a608-b57652636efd)

![image](https://github.com/user-attachments/assets/32e4faf4-9023-4c33-93ee-8cc68939ceb2)

This also affects the "/hassio/system" path. Accessing supervisor logs using the default log book is still working.



So: it's not only, but especially the addon logs.

**Comment by waclaw66** [comment]
Occurs after update Supervisor to 2024.11.2

**Comment by DanGarion** [comment]
Same issue has happened to me after the update to Supervisor 11.2.

**Comment by jacquesChambon** [comment]
Same issue after Supervisor update to 2024.11.2

Core 2024.10.4
Supervisor 2024.11.2
Operating System 13.2
Interface utilisateur 20241002.4

**Comment by kevin-kraus** [comment]
Same Problem here.

Current workaround which is working for me:

1. SSH into HA
2. find docker container of addon using docker ps
3. Copy container ID
4. Use docker logs <container ID> to see the logs

**Comment by thorkseng** [comment]
Try to clean/update the cache of the browser, I am using Edge, and with shift + F5 it works correctly and return the logs.

**Comment by bcutter** [comment]
It's not a frontend cache issue (cleared it). And updating to latest HA Core is also not an option.

It's an issue within Supervisor itself I guess. If it would be dependent of another HA component like Core, the Supervisor 2024.11.0 update would be a breaking change/backward incompatible change/update.

Hopefully a maintainer will have a look at this shortly.

**Comment by neydah700** [comment]
Same issue here, core 2024.11.0 did not fix it for me.

**Comment by fribse** [comment]
I'm also at core 2024.11.0, and no cure.

**Comment by alexsaas** [comment]
Same here since Supervisor Update 2024.11.2

**Comment by wendevlin** [comment]
Hello,
If you are on core 2024.11 it seems to be a caching issue, so clear your cache and hopefully your addon logs works again in the live version.

If you are on < 2024.11 and you don't want to update you need to wait on a fix. I will work on it now.

**Comment by jacquesChambon** [comment]
For me, core update 2024.11.0 fixed the issue

**Comment by alexsaas** [comment]
> If you are on < 2024.11 and you don't want to update you need to wait on a fix. I will work on it now.

That sounds good. It would be great if that could work out in the short term!

**Comment by FelixGaebler** [comment]
![{6AA0176F-C6B8-4626-8B54-0235C08E38E6}](https://github.com/user-attachments/assets/ba3f7c30-eb3c-4c41-bbc9-dfc9b555cb66)

```
Core
2024.10.4
Supervisor
2024.11.2
Operating System
13.2
Frontend
20241002.4
```

same here

**Comment by agners** [comment]
The bugfix will need a new Supervisor release. So I reopen this until then.

**Comment by MRobi1** [comment]
I know this is closed, but is there an ETA on the supervisor release that will fix this?
Having some issues with an addon and would love to be able to view logs.

**Comment by wendevlin** [comment]
> I know this is closed, but is there an ETA on the supervisor release that will fix this?
> 
> Having some issues with an addon and would love to be able to view logs.

It was released last week. You can always see the logs in settings/system/logs and select on the top right your addon.

**Comment by MRobi1** [comment]
> It was released last week.
Well if it was released last week...... I'd say there's still an issue

![image](https://github.com/user-attachments/assets/41d6c6e0-251e-49ff-b296-e5f48eb6c421)

This is going into the addon and clicking logs
![image](https://github.com/user-attachments/assets/0acd7fb8-a142-4b3d-813e-e2b4e3f32b77)
![image](https://github.com/user-attachments/assets/0685827f-7e8d-4c58-8297-340eefbe1dad)

> You can always see the logs in settings/system/logs and select on the top right your addon.
And this is going settings -> System -> Logs and selecting the addon in the top right
![image](https://github.com/user-attachments/assets/8ebf1a3c-b3c4-45a9-9bb9-0a1928ae18c3)
![image](https://github.com/user-attachments/assets/c8fdf14b-b010-472d-a52c-363512c351cc)

My system shows no available updates. These are taken in incognito to be sure it's not a browser cache issue. Is there something I'm missing?

**Comment by wendevlin** [comment]
> EDIT: Just purged cache through cloudflare as well to make sure it's not a cache issue server side. Same results.

So you have a cloudflare tunnel?
Does it also happen when you connect locally?
I would say this is a seperate issue, raise another issue, where you explain everything in detail, how you connect, checkout the browser inspect for network what the actual request issue is.

**Comment by DanGarion** [comment]
This issue only started when the live logs were added to HA over the past 2 months. It seems to be an issue with accessing Addon logs while accessing HA through a Reverse Proxy. If I access direct on my network using local IP or local hostname it works fine, but when using my reverse proxy URL the live logs hang and eventually fail.

I've documented this on the Discover channel without little traction.
There is also a Forum post on this https://community.home-assistant.io/t/live-logs-and-reverse-proxy/801448

I've looked at websocket support and that isn't it. I've tried CORS stuff (as suggested by the SWAG team) and that isn't it. Can we just have the old non live logs back or get someone to look at what is going on? I've looked at all the logs I have access to in HA and in SWAGs nginx and there is nothing.

**Comment by bcutter** [comment]
Addon logs fine and so is `/config/logs`. For me only `/hassio/system` is still broken.

![grafik](https://github.com/user-attachments/assets/28b2a647-4f57-447e-b11c-a2da031eaa01)

Maybe HA Core needs an update or this page is not supported anymore at all.

**Comment by lightelfua** [comment]
Same situation
I think this frontend bug, logs visible in response
 ![Screenshot_2025-01-05_15-56-12](https://github.com/user-attachments/assets/19aac3f0-0648-44d7-8343-0b6e45849fd5)

**Comment by fabiosci** [comment]
I have the same "Loading full log" issue, regardless the browser used (I tried Chrome, Edge and Chrome private mode) and how I access it (homeassistant.local:8123 or IP address).
the issue happens for all addons and from both the addon --> log page and system --> log page (here the only working one is the home assistant core log).
In my case, no errors are showed.

The companion app works correctly and all logs are loaded.
For me the issue started long time ago and the update didn't solve it. My current versions are:

Core 2025.1.0
Supervisor 2024.12.3
Operating System 14.1
Frontend 20250103.0

I'm running home assistant in virtualbox VM

**Comment by MRobi1** [comment]
> So you have a cloudflare tunnel? Does it also happen when you connect locally? I would say this is a seperate issue, raise another issue, where you explain everything in detail, how you connect, checkout the browser inspect for network what the actual request issue is.

No cloudflare tunnel. Cloudflare just hosts my FQDN. I use NGINXPM internally as a reverse proxy. 
If I access via the internal IP address, I can view the logs. If I access via FQDN, I can not.

I believe my issue is related to it being through a reverse proxy which is [this issue here](https://github.com/home-assistant/supervisor/issues/5509).

**Comment by kellerza** [comment]
My logs are also broken. Same error through reverse proxy and direct IP on the internal network

From the frontend, I get a "500 Internal Server error", for both these http calls
  - /api/hassio/addons/core_nginx_proxy/logs
  - /api/hassio/addons/4bb530c7_hass-addon-sunsynk-multi/logs/follow?lines=100

The logs are working OK through CLI

```
docker ps
docker logs -f --tail 50 addon_core_nginx_proxy
```

- Supervisor container log (through CLI)
    ```bash
    docker logs -f --tail 50 hassio_supervisor

    2025-01-11 05:12:03.672 ERROR (MainThread) [aiohttp.server] Error handling request
    Traceback (most recent call last):
      File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 1065, in write
        n = self._sock.send(data)
            ^^^^^^^^^^^^^^^^^^^^^
    BrokenPipeError: [Errno 32] Broken pipe
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 480, in _handle_request
        resp = await request_handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 569, in _handle
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 117, in impl
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 199, in block_bad_requests
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 215, in system_validation
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 285, in token_validation
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 298, in core_proxy
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/utils.py", line 104, in wrap_api
        msg_data = await method(api, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/__init__.py", line 529, in get_addon_logs
        return await self._api_host.advanced_logs(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/utils.py", line 104, in wrap_api
        msg_data = await method(api, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/host.py", line 278, in advanced_logs
        return await self.advanced_logs_handler(request, identifier, follow)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/host.py", line 250, in advanced_logs_handler
        async with self.sys_host.logs.journald_logs(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/contextlib.py", line 210, in __aenter__
        return await anext(self.gen)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/host/logs.py", line 171, in journald_logs
        async with session.get(
                   ^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 1425, in __aenter__
        self._resp: _RetType = await self._coro
                               ^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 730, in _request
        await resp.start(conn)
      File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1059, in start
        message, payload = await protocol.read()  # type: ignore[union-attr]
                           ^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/streams.py", line 671, in read
        await self._waiter
    aiohttp.client_exceptions.ClientOSError: [Errno 32] Broken pipe
    2025-01-11 05:12:03.697 ERROR (MainThread) [aiohttp.server] Error handling request
    Traceback (most recent call last):
      File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 1065, in write
        n = self._sock.send(data)
            ^^^^^^^^^^^^^^^^^^^^^
    BrokenPipeError: [Errno 32] Broken pipe
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 480, in _handle_request
        resp = await request_handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 569, in _handle
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 117, in impl
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 199, in block_bad_requests
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 215, in system_validation
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 285, in token_validation
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 298, in core_proxy
        return await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/utils.py", line 104, in wrap_api
        msg_data = await method(api, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/__init__.py", line 529, in get_addon_logs
        return await self._api_host.advanced_logs(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/utils.py", line 104, in wrap_api
        msg_data = await method(api, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/host.py", line 278, in advanced_logs
        return await self.advanced_logs_handler(request, identifier, follow)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/api/host.py", line 250, in advanced_logs_handler
        async with self.sys_host.logs.journald_logs(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/contextlib.py", line 210, in __aenter__
        return await anext(self.gen)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/usr/src/supervisor/supervisor/host/logs.py", line 171, in journald_logs
        async with session.get(
                   ^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 1425, in __aenter__
        self._resp: _RetType = await self._coro
                               ^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 730, in _request
        await resp.start(conn)
      File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1059, in start
        message, payload = await protocol.read()  # type: ignore[union-attr]
                           ^^^^^^^^^^^^^^^^^^^^^
      File "/usr/local/lib/python3.12/site-packages/aiohttp/streams.py", line 671, in read
        await self._waiter
    aiohttp.client_exceptions.ClientOSError: [Errno 32] Broken pipe
    ```

Version
```
Core 2025.1.2
Supervisor 2024.12.3
Operating System 14.1
Frontend 20250109.0
```
